itest/lnd_custom_features.go

```diff
@@ -0,0 +1,92 @@
+package itest
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/peersrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/require"
+)
+
+// testCustomFeatures tests advertisement of custom features in various bolt 9
+// sets. For completeness, it also asserts that features aren't set in places
+// where they aren't intended to be.
+func testCustomFeatures(ht *lntest.HarnessTest) {
```

Comment:
Nice this itest 🎉.

Read this part of the spec:

> Note that for feature flags which are available in both the node_announcement and [BOLT 11](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md) invoice contexts, the features as set in the [BOLT 11](https:// invoice should override those set in the node_announcement. This keeps things consistent with the unknown features behavior as specified in [BOLT 7](https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#the-node_announcement-message).

Does this mean that there is a case where we have to update the node announcement feature set when the invoice feature set is updated and this feature happens to be present in the node announcement ? Wondering if this is considered in the codebase in gerneral and we need to take care when we update a feature via RPC?

Reply:
> there is a case where we have to update the node announcement feature set when the invoice feature set is updated and this feature happens to be present in the node announcement

No, I think it's getting at what to do when a feature bit _can_ bet set in both of those contexts, but they say diffierent things.

For example: if a node advertises feature X in its announcement as optional, but then its invoice says that X is required in the invoice then it should be interpreted as required?

Reply:
A ok so no feature needs to be overwritten but only the invoice feature counts if for example the X feature is set differently in the node_announcement vs in the invoice feature context. Wording in the spec with `should override` was a bit weird.

Reply:
maybe I write an issue in the bolt repo to be more clear about it ?

Reply:
I think the wording is clear enough fwiw.